src/mon/OSDMonitor.cc

```diff
@@ -6036,21 +6036,39 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
     } else {
       if (f)
 	f->open_array_section("pools");
-      for (map<int64_t,pg_pool_t>::const_iterator it = osdmap.get_pools().begin();
-	   it != osdmap.get_pools().end();
-	   ++it) {
+      for (auto &[pid, pdata] : osdmap.get_pools()) {
 	if (f) {
 	  if (detail == "detail") {
 	    f->open_object_section("pool");
-	    f->dump_int("pool_id", it->first);
-	    f->dump_string("pool_name", osdmap.get_pool_name(it->first));
-	    it->second.dump(f.get());
+	    f->dump_int("pool_id", pid);
+	    f->dump_string("pool_name", osdmap.get_pool_name(pid));
+	    pdata.dump(f.get());
+	    if (pdata.get_type() == pg_pool_t::TYPE_REPLICATED) {
+	      // Add wlb section with values for score, optimal score, raw score
+	      // and primary_affinity average
+	      OSDMap::wl_balance_info_t wlb_info;
+	      f->open_object_section("workload_balance");
+	      f->dump_float("score (acting)",
+	        osdmap.calc_wl_balance_score(g_ceph_context, pid, &wlb_info));
```

Comment:
What you need to do here is either pass in `cct` as the context, or, at the top of the file, include this line: if you want to use the global context:

`#define dout_context g_ceph_context`

When I grepped for "g_ceph_context" under src, I found a ton of examples, such as in `mgr/DaemonServer.cc`, where it is defined.